app/services/make_user_a_mentor.rb

```diff
@@ -0,0 +1,35 @@
+class MakeUserAMentor
+  include Mandate
+
+  initialize_with :user, :track_id
+
+  def call
+    make_mentor
+    create_track_mentorship
+    invite_to_slack
+  end
+
+  def make_mentor
+    user.update(is_mentor: true)
+  end
+
+  def create_track_mentorship
+    user.track_mentorships.create(track_id: track_id)
+  end
+
+  def invite_to_slack
+    data = {
```

Comment:
Is there a slack ruby gem we could use? Just thinking it might lessen maintenance on this part of code.

Reply:
I also think we should put this into a separate Sidekiq job in order for it to automatically retry if ever we hit rate limit errors from Slack.

Reply:
Good ideas. The Slack Gem doesn't support the endpoint I'm using (it's undocumented - **sigh**). I think in the long-run we probably get people to click a link to get into Slack, rather than getting the invite email.